dialogflow/stopwatch/Stopwatch/DialogflowViewController.swift

```diff
@@ -0,0 +1,320 @@
+//
+//  DialogflowViewController.swift
+//  DFSampleUI
+//
+//  Created by Santhosh Vaddi on 1/20/19.
+//  Copyright © 2019 Santhosh Vaddi. All rights reserved.
+//
+
+import UIKit
+import MaterialComponents
+import AVFoundation
+import googleapis
+
+
+let selfKey = "Self"
+let botKey = "Bot"
+
+class DialogflowViewController: UIViewController {
+    var appBar = MDCAppBar()
+    var audioData: NSMutableData!
+    var listening: Bool = false
+    var isFirst: Bool = true
+    // Text Field
+    var intentTextField: MDCTextField = {
+        let textField = MDCTextField()
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        return textField
+    }()
+    var  textFieldBottomConstraint: NSLayoutConstraint!
+    let intentTextFieldController: MDCTextInputControllerOutlined
+    var tableViewDataSource = [[String: String]]()
+    
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var optionsCard: MDCCard!
```

Comment:
I noticed that this card has a nice animated response to touches, but it isn't actually triggering any action. That seems weird when we are displaying the cancel button. I also think that it would be nice for the cancel button to be centered. 

Looking more closely at the Assistant app, It appears that when the microphone button is pressed, the keyboard button is hidden and the microphone button is moved to the center and changed to show the listening animation. Touching the listening animation cancels listening and undoes these changes. I think that behavior might be fairly easy to add and would be a nice improvement to to the user experience.

Reply:
I agree, I have made the necessary changes to the card and now we are able to move the cancel button to center after tapping on the microphone button.